x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/predicate/operator/comparison/EsqlBinaryComparison.java

```diff
@@ -49,13 +104,19 @@ protected EsqlBinaryComparison(
         Source source,
         Expression left,
         Expression right,
-        BinaryComparisonProcessor.BinaryComparisonOperation operation,
+        BinaryComparisonOpeartion operation,
         // TODO: We are definitely not doing the right thing with this zoneId
         ZoneId zoneId,
         Map<DataType, EsqlArithmeticOperation.BinaryEvaluator> evaluatorMap
     ) {
-        super(source, left, right, operation, zoneId);
+        super(source, left, right, operation.shim, zoneId);
+        assert zoneId == null;
```

Comment:
Turns out at least some of our tests pass a non-null zoneId.  I need to dig into this more and see if that's just coverage, or if we expect some behavior out of that.